hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorPartition.java

```diff
@@ -73,6 +81,23 @@ public Operation prepareReplicationOperation(int replicaIndex, boolean migration
         return new ReplicationOperation(map);
     }
 
+    public Map<String, Collection<ScheduledTaskDescriptor>> prepareOwnedSnapshot() {
+        Map<String, Collection<ScheduledTaskDescriptor>> containers = new HashMap<String, Collection<ScheduledTaskDescriptor>>();
+
+        if (logger.isFinestEnabled()) {
+            logger.finest("[Partition: " + partitionId + "] Prepare snapshot of partition owned tasks.");
+        }
+
+        for (ScheduledExecutorContainer container : getContainers()) {
+            SplitBrainMergePolicy mergePolicy = getMergePolicy(container.getName());
+            if (!(mergePolicy instanceof DiscardMergePolicy)) {
+                containers.put(container.getName(), container.prepareForReplication(true).values());
```

Comment:
Can this method throw exceptions indefinitely (e.g. [here](https://github.com/hazelcast/hazelcast/blob/c64de8011f30e321fca0196e2e7769f45fb9ba51/hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java#L327-L337))? This means migrations might fail and be retried indefinitely.

Reply:
TBH, haven't considered this. I will guard it accordingly.